### ISSUE TITLE
Different versions of the project have different training results.

### ISSUE THREAD

**Issue by nezcs** [ultralytics/ultralytics#23248]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hello author!
Thank you for releasing the latest yolo26 model.
I found a problem: my original project version was ultralytics 8.3.236, and the latest version is ultralytics 8.4.0. These two projects train yolo11 model under the conditions of parameters and equipment (including gpu). map50 and map50-95 are very different under the same data set, for example, (map50 in version 8.3.236 is 88%, while map50 in version 8.4.0 is only 82%). Is this normal?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @nezcs, thank you for your interest in Ultralytics 🚀! This is an automated response to help us triage your issue quickly—an Ultralytics engineer will also assist soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question (like comparing `ultralytics==8.3.236` vs `8.4.0` results on YOLO11), please share enough details for us to reproduce and compare accurately, including:
-Exact command/config used (CLI or Python)
-Your `data.yaml` (or a sanitized version)
-A few representative dataset image samples
-Training logs for both runs (full console output if possible)
-Any seed settings you used (if any)
-Model used (e.g., YOLO11 variant) and all training args

Also, please verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) ✅

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
How many images do you have?

**Comment by nezcs** [comment]
> How many images do you have?你有多少张图片？

12000张

**Comment by Y-T-G** [comment]
Some of the default parameters have changed. So you can compare the args.yaml from older versions with newer versions and keep them the same in newer version

**Comment by nezcs** [comment]
> Some of the default parameters have changed. So you can compare the args.yaml from older versions with newer versions and keep them the same in newer version一些默认参数已经改变。所以你可以比较旧版本的 args.yaml，并在新版本中保持不变

我还发现了一个问题,我的环境中超lytics为8.4.0,但是在训练中为什么打印以下:Ultralytics 8.3.183
(yolox)root @ autodl-container-40fd 419 ed 7-2d 4c da 86:~ # python/root/autodl-tmp/ultralytics-main-old/train . py
Ultralytics🚀python-3 . 10 . 19 torch-2 . 2 . 0+Cu 118 CUDA:0(英伟达GeForce RTX 4090 D，24081MiB)

**Comment by Y-T-G** [comment]
Because you have a local Ultralytics folder that uses an older version. The local folder takes priority

**Comment by Y-T-G** [comment]
The new version also updates this layer with `act=False`. So you can change that if you want to. Previously it had `act=True`

https://github.com/ultralytics/ultralytics/blob/fff04212c7b9d58c15d09c24ea9ebc5ca35ad0cb/ultralytics/nn/modules/block.py#L226

**Comment by GKXKX123456** [comment]
I am encountering the same issue. I trained YOLOv11s using the exact same dataset and training args, but saw a significant regression:

- v8.3.63: mAP50 reached 87.9%
- v8.3.237: mAP50 reached 86.4%
- v8.4.2: mAP50 dropped to just 55.0%

I find this behavior very strange.

**Comment by nezcs** [comment]
> I am encountering the same issue. I trained YOLOv11s using the exact same dataset and training args, but saw a significant regression:我也遇到了同样的问题。我用完全相同的数据集和训练 args 训练了 YOLOv11，但发现了显著的回归：
> 
> * v8.3.63: mAP50 reached 87.9%v8.3.63：mAP50 达到 87.9%
> * v8.3.237: mAP50 reached 86.4%v8.3.237：mAP50 达到 86.4%
> * v8.4.2: mAP50 dropped to just 55.0%v8.4.2：mAP50 降至仅 55.0%。
> 
> I find this behavior very strange.我觉得这种行为非常奇怪。

You can try the latest version. I encountered this problem in 8.4.2. When I downloaded the 8.4.3 version again, it solved the problem.

**Comment by GKXKX123456** [comment]
> > 我遇到了同样的问题。我使用完全相同的数据集和训练参数训练 YOLOv11，但看到了显着的回归：我也遇到了同样的问题。我用了 4 个数据集和训练参数训练了 YOLOv11，但发现了明显的回归：
> > 
> > * v8.3.63：mAP50达到87.9%v8.3.63：mAP50达到87.9%
> > * v8.3.237：mAP50达到86.4%v8.3.237：mAP50达到86.4%
> > * v8.4.2: mAP50 降至仅 55.0%v8.4.2：mAP50 仅 55.0%。
> > 
> > 我觉得这种行为很奇怪。我觉得这种行为非常奇怪。
> 
> 您可以试试最新版本。我在 8.4.2 版本中遇到过这个问题。当我重新下载 8.4.3 版本后，问题就解决了。

Thanks, I'll try that.

**Comment by GKXKX123456** [comment]
> > > 我遇到了同样的问题。我使用完全相同的数据集和训练参数训练 YOLOv11，但看到了显着的回归：我也遇到了同样的问题。我用了 4 个数据集和训练参数训练了 YOLOv11，但发现了明显的回归：
> > > 
> > > * v8.3.63：mAP50达到87.9%v8.3.63：mAP50达到87.9%
> > > * v8.3.237：mAP50达到86.4%v8.3.237：mAP50达到86.4%
> > > * v8.4.2: mAP50 降至仅 55.0%v8.4.2：mAP50 仅 55.0%。
> > > 
> > > 我觉得这种行为很奇怪。我觉得这种行为非常奇怪。
> > 
> > 
> > 您可以试试最新版本。我在 8.4.2 版本中遇到过这个问题。当我重新下载 8.4.3 版本后，问题就解决了。
> 
> Thanks, I'll try that.

The issue is resolved after updating to v8.4.6. The training mAP is back to normal now. Thank you very much!

**Comment by glenn-jocher** [comment]
Good to hear it’s back to normal on `v8.4.6`—to keep results reproducible, pin that version and confirm you’re training against the intended install (no local clone shadowing) by running:

```bash
yolo checks
```

If you see another regression, please open a new issue with the full training command, complete logs, and the `yolo checks` output; the [Troubleshooting Common YOLO Issues guide](https://docs.ultralytics.com/guides/yolo-common-issues/) also covers the most common env/version pitfalls.
